### ISSUE TITLE
Self-hosted router repeats error bad argument in call to erlang:list_to_integer([]) router_sc_worker:get_sc_buffer:395

### ISSUE THREAD

**Issue by mfalkvidd** [helium/router#407]
My self-hosted router repeats this error every 15 seconds. The router seems to work, but having errors in the log seems liek a bad thing.
```
[2021-08-17 12:04:45.353] <0.27303.0> [error] [router_sc_worker:get_sc_buffer:395] gen_server router_sc_worker terminated with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:04:45.353] <0.27303.0> [error] [router_sc_worker:get_sc_buffer:395] CRASH REPORT Process router_sc_worker with 0 neighbours crashed with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:04:45.353] <0.1477.0> [error] [router_sc_worker:get_sc_buffer:395] Supervisor router_sup had child router_sc_worker started with router_sc_worker:start_link(#{}) at <0.27303.0> exit with reason bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395 in context child_terminated
[2021-08-17 12:05:00.364] <0.27330.0> [error] [router_sc_worker:get_sc_buffer:395] gen_server router_sc_worker terminated with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:00.364] <0.27330.0> [error] [router_sc_worker:get_sc_buffer:395] CRASH REPORT Process router_sc_worker with 0 neighbours crashed with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:00.365] <0.1477.0> [error] [router_sc_worker:get_sc_buffer:395] Supervisor router_sup had child router_sc_worker started with router_sc_worker:start_link(#{}) at <0.27330.0> exit with reason bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395 in context child_terminated
[2021-08-17 12:05:15.373] <0.27350.0> [error] [router_sc_worker:get_sc_buffer:395] gen_server router_sc_worker terminated with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:15.373] <0.27350.0> [error] [router_sc_worker:get_sc_buffer:395] CRASH REPORT Process router_sc_worker with 0 neighbours crashed with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:15.374] <0.1477.0> [error] [router_sc_worker:get_sc_buffer:395] Supervisor router_sup had child router_sc_worker started with router_sc_worker:start_link(#{}) at <0.27350.0> exit with reason bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395 in context child_terminated
[2021-08-17 12:05:30.380] <0.27369.0> [error] [router_sc_worker:get_sc_buffer:395] gen_server router_sc_worker terminated with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:30.380] <0.27369.0> [error] [router_sc_worker:get_sc_buffer:395] CRASH REPORT Process router_sc_worker with 0 neighbours crashed with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:30.381] <0.1477.0> [error] [router_sc_worker:get_sc_buffer:395] Supervisor router_sup had child router_sc_worker started with router_sc_worker:start_link(#{}) at <0.27369.0> exit with reason bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395 in context child_terminated
[2021-08-17 12:05:45.389] <0.27432.0> [error] [router_sc_worker:get_sc_buffer:395] gen_server router_sc_worker terminated with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:45.389] <0.27432.0> [error] [router_sc_worker:get_sc_buffer:395] CRASH REPORT Process router_sc_worker with 0 neighbours crashed with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:05:45.390] <0.1477.0> [error] [router_sc_worker:get_sc_buffer:395] Supervisor router_sup had child router_sc_worker started with router_sc_worker:start_link(#{}) at <0.27432.0> exit with reason bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395 in context child_terminated
[2021-08-17 12:06:00.396] <0.27453.0> [error] [router_sc_worker:get_sc_buffer:395] gen_server router_sc_worker terminated with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:06:00.396] <0.27453.0> [error] [router_sc_worker:get_sc_buffer:395] CRASH REPORT Process router_sc_worker with 0 neighbours crashed with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:06:00.397] <0.1477.0> [error] [router_sc_worker:get_sc_buffer:395] Supervisor router_sup had child router_sc_worker started with router_sc_worker:start_link(#{}) at <0.27453.0> exit with reason bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395 in context child_terminated
```

Complete log around one of the events:
```
[2021-08-17 12:09:10.094] <0.1546.0> [info] [router_metrics:handle_info:200] running metrcis
[2021-08-17 12:09:15.542] <0.27829.0> [info] [router_sc_worker:maybe_start_state_channel:271] [active: 1] [opened: 1] [in flight 0] [max: 15] all active, opening more
[2021-08-17 12:09:15.547] <0.1516.0> [warning] [blockchain_event:terminate:108] terminating remove_handler
[2021-08-17 12:09:15.548] <0.27829.0> [error] [router_sc_worker:get_sc_buffer:395] gen_server router_sc_worker terminated with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:09:15.548] <0.27829.0> [error] [router_sc_worker:get_sc_buffer:395] CRASH REPORT Process router_sc_worker with 0 neighbours crashed with reason: bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395
[2021-08-17 12:09:15.548] <0.1477.0> [error] [router_sc_worker:get_sc_buffer:395] Supervisor router_sup had child router_sc_worker started with router_sc_worker:start_link(#{}) at <0.27829.0> exit with reason bad argument in call to erlang:list_to_integer([]) in router_sc_worker:get_sc_buffer/0 line 395 in context child_terminated
[2021-08-17 12:09:15.549] <0.1516.0> [warning] [blockchain_event:terminate:108] terminating remove_handler
[2021-08-17 12:09:15.549] <0.1516.0> [warning] [blockchain_event:terminate:108] terminating remove_handler
[2021-08-17 12:09:15.549] <0.27832.0> [warning] [router_console_ws_worker:terminate:289] went down shutdown
[2021-08-17 12:09:15.551] <0.27848.0> [info] [router_sc_worker:init:80] router_sc_worker init with #{}
[2021-08-17 12:09:15.551] <0.27850.0> [info] [router_console_api:init:385] router_console_api init with #{downlink_endpoint => <<>>,endpoint => <<"http://console:4000">>,secret => <<"<deleted>">>,ws_endpoint => <<"ws://console:4000/socket/router/websocket">>}
[2021-08-17 12:09:15.558] <0.27851.0> [info] [router_console_ws_worker:init:51] router_console_ws_worker init with #{downlink_endpoint => <<>>,endpoint => <<"http://console:4000">>,secret => <<"<deleted>">>,ws_endpoint => <<"ws://console:4000/socket/router/websocket">>}
[2021-08-17 12:09:15.558] <0.27852.0> [info] [router_console_ws_handler:init:78] router_console_ws_handler init with #{auto_join => [<<"device:all">>,<<"organization:all">>,<<"label:all">>],forward => <0.27851.0>,url => "ws://console:4000/socket/router/websocket?token=<deleted>&vsn=2.0.0"}
[2021-08-17 12:09:15.558] <0.27853.0> [info] [router_console_dc_tracker:init:150] router_console_dc_tracker init with #{}
[2021-08-17 12:09:15.559] <0.27854.0> [info] [router_v8:init:57] router_v8 init with #{}
[2021-08-17 12:09:15.559] <0.27857.0> [info] [router_device_devaddr:init:139] router_device_devaddr init with #{}
[2021-08-17 12:09:15.560] <0.27859.0> [info] [router_xor_filter_worker:init:168] router_xor_filter_worker init with #{}
[2021-08-17 12:09:15.562] <0.27851.0> [info] [router_console_ws_worker:handle_info:81] joined, sending router address to console
[2021-08-17 12:09:15.563] <0.27851.0> [warning] [router_console_ws_worker:handle_info:282] rcvd unknown info msg: {ws_message,<<"device:all">>,<<"phx_reply">>,#{<<"response">> => #{},<<"status">> => <<"ok">>}}, {state,<0.27852.0>,<<"ws://console:4000/socket/router/websocket">>,#Ref<0.3690382999.2350776331.225163>,#Ref<0.3690382999.2350776331.225165>}
[2021-08-17 12:09:15.563] <0.27851.0> [warning] [router_console_ws_worker:handle_info:282] rcvd unknown info msg: {ws_message,<<"organization:all">>,<<"phx_reply">>,#{<<"response">> => #{},<<"status">> => <<"ok">>}}, {state,<0.27852.0>,<<"ws://console:4000/socket/router/websocket">>,#Ref<0.3690382999.2350776331.225163>,#Ref<0.3690382999.2350776331.225165>}
[2021-08-17 12:09:15.564] <0.27851.0> [warning] [router_console_ws_worker:handle_info:282] rcvd unknown info msg: {ws_message,<<"label:all">>,<<"phx_reply">>,#{<<"response">> => #{},<<"status">> => <<"ok">>}}, {state,<0.27852.0>,<<"ws://console:4000/socket/router/websocket">>,#Ref<0.3690382999.2350776331.225163>,#Ref<0.3690382999.2350776331.225165>}
[2021-08-17 12:09:15.564] <0.27851.0> [warning] [router_console_ws_worker:handle_info:282] rcvd unknown info msg: {ws_message,<<"organization:all">>,<<"phx_reply">>,#{<<"response">> => #{},<<"status">> => <<"ok">>}}, {state,<0.27852.0>,<<"ws://console:4000/socket/router/websocket">>,#Ref<0.3690382999.2350776331.225163>,#Ref<0.3690382999.2350776331.225165>}
[2021-08-17 12:09:16.067] <0.27859.0> [info] [router_xor_filter_worker:handle_info:361] filters are still up to date, devices have maybe have changed, updating cache
```

The problem might be related to the following error from my Console:
```
12:12:31.236 [error] #PID<0.16272.0> running ConsoleWeb.Endpoint (connection #PID<0.16264.0>, stream id 3) terminated
Server: console:4000 (http)
Request: POST /api/router/devices/update_in_xor_filter
** (exit) an exception was raised:
    ** (RuntimeError) expected action/2 to return a Plug.Conn, all plugs must receive a connection (conn) and return a connection, got: nil
        (console) lib/console_web/controllers/router/device_controller.ex:1: ConsoleWeb.Router.DeviceController.phoenix_controller_pipeline/2
        (phoenix) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (console) lib/console_web/endpoint.ex:1: ConsoleWeb.Endpoint.plug_builder_call/2
        (console) lib/console_web/endpoint.ex:1: ConsoleWeb.Endpoint.call/2
        (phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:65: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy) /app/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
        (cowboy) /app/deps/cowboy/src/cowboy_stream_h.erl:320: :cowboy_stream_h.execute/3
        (cowboy) /app/deps/cowboy/src/cowboy_stream_h.erl:302: :cowboy_stream_h.request_process/3
12:12:45.721 [info] POST /api/router/sessions
12:12:45.722 [info] Sent 201 in 521µs
12:12:45.726 [info] CONNECTED TO ConsoleWeb.RouterSocket in 372µs
  Transport: :websocket
  Serializer: Phoenix.Socket.V2.JSONSerializer
  Parameters: %{"token" => "<deleted>", "vsn" => "2.0.0"}
12:12:45.728 [info] JOINED device:all in 22µs
  Parameters: %{}
12:12:45.729 [info] JOINED organization:all in 16µs
  Parameters: %{}
12:12:45.730 [info] JOINED label:all in 13µs
  Parameters: %{}
12:12:46.229 [info] GET /api/router/devices
12:12:46.231 [info] Sent 200 in 1ms
12:12:46.236 [info] POST /api/router/devices/update_in_xor_filter
12:12:46.237 [info] Sent 200 in 665µs
12:12:46.240 [error] #PID<0.16281.0> running ConsoleWeb.Endpoint (connection #PID<0.16273.0>, stream id 3) terminated
Server: console:4000 (http)
Request: POST /api/router/devices/update_in_xor_filter
** (exit) an exception was raised:
    ** (RuntimeError) expected action/2 to return a Plug.Conn, all plugs must receive a connection (conn) and return a connection, got: nil
        (console) lib/console_web/controllers/router/device_controller.ex:1: ConsoleWeb.Router.DeviceController.phoenix_controller_pipeline/2
        (phoenix) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (console) lib/console_web/endpoint.ex:1: ConsoleWeb.Endpoint.plug_builder_call/2
        (console) lib/console_web/endpoint.ex:1: ConsoleWeb.Endpoint.call/2
        (phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:65: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy) /app/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
        (cowboy) /app/deps/cowboy/src/cowboy_stream_h.erl:320: :cowboy_stream_h.execute/3
        (cowboy) /app/deps/cowboy/src/cowboy_stream_h.erl:302: :cowboy_stream_h.request_process/3
12:13:00.733 [info] POST /api/router/sessions
```

**Comment by macpie** [comment]
@mfalkvidd thanks for the report. There was a bug when a value was not set.

Also as we add new configuration we usually put it in https://github.com/helium/router/blob/master/.env-template#L21

**Comment by mfalkvidd** [comment]
Excellent, thanks. I'll add the value tomorrow and report back.

**Comment by macpie** [comment]
Issue fixed  in #409

**Comment by mfalkvidd** [comment]
Adding the value got rid of the error messages in router. Thanks!
